string obj
int16 tag
